src/wms/src/test/java/org/geoserver/wms/map/GetMapXmlReaderTest.java

```diff
@@ -132,6 +137,32 @@ public void testAllowDynamicStyles() throws Exception {
         }
     }
 
+    @Test
+    public void testCleanServiceException() throws Exception {
```

Comment:
the clean exception code has 2 major paths - SAXException and IO Exception.  Should probably have test for both cases.  

Also, the SAXException cleaning might want to have more test cases (see above comment - "if" - in implementation)

Reply:
Added test case for this to DispatcherTest which tests XmlRequestReader using a dummy service.